workflow-controller/controller-service/src/main/java/io/hyscale/controller/util/ServiceSpecMapper.java

```diff
@@ -65,16 +66,23 @@ public ServiceSpec from(File serviceSpecFile) throws HyscaleException {
     }
 
     public ServiceSpec from(String filepath) throws HyscaleException {
+        if (StringUtils.isBlank(filepath)) {
+            throw getException(filepath);
+        }
         File serviceSpecFile = new File(filepath);
         checkForFile(serviceSpecFile);
         return from(serviceSpecFile);
     }
 
     private boolean checkForFile(File serviceSpecFile) throws HyscaleException {
         if (serviceSpecFile == null || !serviceSpecFile.exists()) {
-            throw new HyscaleException(ControllerErrorCodes.CANNOT_FIND_SERVICE_SPEC,
-                    serviceSpecFile != null ? serviceSpecFile.getName() : ToolConstants.EMPTY_STRING);
+            throw getException(serviceSpecFile != null ? serviceSpecFile.getName() : ToolConstants.EMPTY_STRING);
         }
         return true;
     }
+
+    private HyscaleException getException(String serviceSpec) {
+        return new HyscaleException(ControllerErrorCodes.CANNOT_FIND_SERVICE_SPEC,
+                serviceSpec != null ? serviceSpec : ToolConstants.EMPTY_STRING);
```

Comment:
This method getException is not  satisfying it's name , because you aren't fetching but you are building .